packages/slate-editor/src/extensions/paste-slate-content/lib/withSlatePasting.ts

```diff
@@ -38,3 +44,14 @@ function withoutSlateFragmentData(data: DataTransfer): DataTransfer {
     const dataMap = Object.fromEntries(types.map((type) => [type, data.getData(type)]));
     return createDataTransfer(dataMap);
 }
+
+function handlePastingIntoImageCaption(editor: Editor, fragment: SlateFragment) {
+    const [imageNode] = Array.from(Editor.nodes(editor, { match: isImageNode })).at(0) ?? [];
+
+    if (imageNode && EditorCommands.isNodeEmpty(editor, imageNode)) {
+        Transforms.insertNodes(editor, fragment, { at: editor.selection?.anchor.path });
```

Comment:
Passing explicit path does the trick and this code is skipped https://github.com/ianstormtaylor/slate/blob/main/packages/slate/src/transforms/node.ts#L196 and therefore new node is inserted into image caption node

Reply:
I dislike the fact that the generic code of `withSlatePasting` started to be dependent on our model -- specifically it's aware of our ImageNodes. That's a code smell to me.

Is there a way to avoid hard-coding it to use special logic just for images? Why images are special? Can we change it around by checking for `editor.isVoid` or `editor.isRichBlock`?

Reply:
> Why images are special?
Because they can be technically empty as the children is empty, and at the same visually is not empty as there is an image. When you past smth into empty paragraphs it will be replaced with new content and it feels expected, but not for images.

I just noticed that we have the same situation for quotes, agree with general solution. Maybe we could add special callback into Extension and check it here?

Reply:
> Because they can be technically empty as the children is empty, and at the same visually is not empty as there is an image. When you past smth into empty paragraphs it will be replaced with new content and it feels expected, but not for images.

I believe that `editor.isRichBlock()` is a method to do exactly this check. Rich blocks, even having empty children, should not be considered empty.